### PR TITLE
feat(*): Upgrade Grafana to v11.3.0

### PR DESCRIPTION
**Pull request by ifrost**
### ✨ Description

**Related issue(s):** `-`

This PR upgrades Grafana to v11.3.0

### 📖 Summary of the changes

All references upgraded from 11.2.0 to 11.3.0

### 🧪 How to test?

- The build should pass

### CODE REVIEW

File: src/shared/ui/PluginInfo.tsx [hunk 1]
```diff
@@ -14,18 +14,24 @@ const { buildInfo: grafanaBuildInfo } = config;
...
-    </div>
-  );
+  if (context) {
+    const {
+      meta: {
+        info: { version, updated },
+      },
+    } = context;
+
+    return (
```
Comment (grafakus) [inline src/shared/ui/PluginInfo.tsx:26]: Good catch.
Instead of removing _entirely_ the header, why not gracefully degrade?
At least by displaying a question mark or "information not available", so it wouldn't be perceived as a bug from the user perspective.

File: src/app/components/Routes/Routes.tsx [hunk 2]
```diff
@@ -13,13 +13,13 @@ export function Routes() {
 
   return (
     <Switch>
-      <Route path={prefixRouteWithPluginBaseUrl(ROUTES.PROFILES_EXPLORER_VIEW)} exact>
```
Comment (grafakus) [inline src/app/components/Routes/Routes.tsx:16]: Why the change?

Comment (ifrost) [reply src/app/components/Routes/Routes.tsx:16]: `exact` param is not available in react router 6:

```
ERROR in ./src/app/components/Routes/Routes.tsx:22:67
TS2322: Type '{ children: Element; path: string; exact: true; }' is not assignable to type 'IntrinsicAttributes & RouteProps'.
  Property 'exact' does not exist on type '(IntrinsicAttributes & PathRouteProps) | (IntrinsicAttributes & LayoutRouteProps)'.
    20 |         <AdHocView />
    21 |       </Route>
  > 22 |       <Route path={prefixRouteWithPluginBaseUrl(ROUTES.SETTINGS)} exact>
       |                                                                   ^^^^^
    23 |         <SettingsView />
    24 |       </Route>
    25 |       {/* Default Route */}
```

from [docs](https://reactrouter.com/en/main/upgrading/v5#relative-routes-and-links):

> `<Route exact>` is gone. Instead, routes with descendant routes (defined in other components) use a trailing * in their path to indicate they match deeply

File: src/shared/ui/PluginInfo.tsx (hunk 1 above)
Comment (ifrost) [reply src/shared/ui/PluginInfo.tsx]: Good idea! How about [this](https://github.com/grafana/explore-profiles/pull/287/commits/0b3f2e1b337e42155c2c650e95383d575b32edb6)?

Screenshot 2024-11-19 at 14 36 24

File: src/app/components/Routes/Routes.tsx (hunk 2 above)
Comment (grafakus) [reply src/app/components/Routes/Routes.tsx:16]: Then we have to prevent discrepancies between the local environment/devx and the other environments by also upgrading React Router in the npm dependencies.

Comment (ifrost) [reply src/app/components/Routes/Routes.tsx]: I have reverted the change and added explicit react-router version in package.json. Grafana 11.3.0 provides v5. Locally if we don't specify react-router it starts pulling v6 to node_modules 🤔

Comment (grafakus) [reply src/app/components/Routes/Routes.tsx]: > it starts pulling v6 to node_modules
Transitive dependencies FTW!

**Review by grafakus — APPROVED** [review]
👍🏾👍🏾
